src/transformStateWithClones.js

```diff
@@ -7,7 +7,31 @@
  * @return {Object[]}
  */
 function transformStateWithClones(state, actions) {
-  // write code here
+  let currentState = { ...state };
+  const setOfStates = [];
+
+  for (const action of actions) {
+    switch (action.type) {
+      case 'clear':
+        currentState = {};
+        setOfStates.push(currentState);
+        break;
+      case 'addProperties':
+        currentState = { ...currentState, ...action.extraData };
```

Comment:
The name `currentState` could be improved to reflect that it is a copy of the state. A better name might be `stateCopy` or `newState`.